Dockerfile

```diff
@@ -59,13 +66,25 @@ COPY --from=goversioninfo /out/goversioninfo /usr/bin/goversioninfo
 RUN [ ! -f /etc/alpine-release ] && xx-info is-cross && [ "$(xx-info arch)" = "arm64" ] && XX_CC_PREFER_LINKER=ld xx-clang --setup-target-triple || true
 RUN --mount=type=bind,target=.,ro \
     --mount=type=cache,target=/root/.cache \
-    --mount=from=dockercore/golang-cross:xx-sdk-extras,target=/xx-sdk,src=/xx-sdk \
-    --mount=type=tmpfs,target=cli/winresources \
-    # override the default behavior of go with xx-go
-    xx-go --wrap && \
-    # export GOCACHE=$(go env GOCACHE)/$(xx-info)$([ -f /etc/alpine-release ] && echo "alpine") && \
-    TARGET=/out ./scripts/build/binary && \
-    xx-verify $([ "$GO_LINKMODE" = "static" ] && echo "--static") /out/docker
+    --mount=from=osxcross,src=/osxcross,target=/osxcross \
+    --mount=from=osxcross,src=/osxsdk,target=/xx-sdk \
+    --mount=type=tmpfs,target=cli/winresources <<EOT
+  set -e
+  if [ "$BASE_VARIANT" != "alpine" ] && [ "$(xx-info os)" = "darwin" ]; then
+    # on debian, use osxcross toolchain that includes the llvm-dsymutil tool
+    # and right minimal llvm version required
+    # more info: https://github.com/docker/cli/pull/3717
+    export PATH="/osxcross/bin:$PATH"
+    export LD_LIBRARY_PATH="/osxcross/lib:$LD_LIBRARY_PATH"
+    export CC="o64-clang"
+    export CXX="o64-clang++"
+  fi
```

Comment:
@tonistiigi If you're ok with `osxcross` we could add a detection folder like `/xx-sdk` on xx so it could set itself the required env vars to use `osxcross` and we could remove this part in the Dockerfile? Maybe also the ability to support `macports` like https://github.com/crazy-max/goxx/blob/main/rootfs/usr/local/bin/goxx-macports. WDYT?